src/cargo/core/compiler/mod.rs

```diff
@@ -1095,6 +1096,12 @@ fn add_allow_features(build_runner: &BuildRunner<'_, '_>, cmd: &mut ProcessBuild
 ///
 /// [`--error-format`]: https://doc.rust-lang.org/nightly/rustc/command-line-arguments.html#--error-format-control-how-errors-are-produced
 fn add_error_format_and_color(build_runner: &BuildRunner<'_, '_>, cmd: &mut ProcessBuilder) {
+    let enable_timings = build_runner.bcx.gctx.nightly_features_allowed
+        && !build_runner.bcx.build_config.timing_outputs.is_empty();
```

Comment:
> Cargo has to somehow figure out if it should pass --json=timings to rustc or not. I don't think that it's worth it to add a new unstable Cargo flag just for this (ofc if you disagree, I'd be happy to implement that), instead I think that Cargo could just detect if the used rustc is nightly, and if yes, pass the flag always. I'm not sure how best to detect that though. So far I just used gctx.nightly_features_allowed, but I don't know if it's correct.

Reply:
Did you mean to reply something here? :sweat_smile: It looks like CI/tests doesn't like using just `nightly_features_allowed`..

Reply:
I was trying to move your comment in the PR description to an issue on the PR so it has a proper place to be discussed and not be missed.